src/algebra/group/basic.lean

```diff
@@ -482,115 +482,137 @@ lemma div_mul_comm (a b c d : G) : a / b * (c / d) = a * c / (b * d) :=
 
-lemma sub_sub_self (a b : G) : a - (a - b) = b :=
-by simpa using add_neg_cancel_left a b
+@[to_additive sub_sub_self]
+lemma div_div_self' (a b : G) : a / (a / b) = b :=
+by simpa using mul_inv_cancel_left a b
 
-lemma add_sub_comm (a b c d : G) : a + b - (c + d) = (a - c) + (b - d) :=
+@[to_additive add_sub_comm]
+lemma mul_div_comm' (a b c d : G) : a * b / (c * d) = (a / c) * (b / d) :=
 by simp
 
-lemma sub_eq_sub_add_sub (a b c : G) : a - b = c - b + (a - c) :=
-begin simp, rw [add_left_comm c], simp end
+@[to_additive]
+lemma div_eq_div_mul_div (a b c : G) : a / b = c / b * (a / c) :=
+begin simp, rw [mul_left_comm c], simp end
 
-lemma neg_neg_sub_neg (a b : G) : - (-a - -b) = a - b :=
+@[to_additive]
+lemma inv_inv_div_inv (a b : G) : (a⁻¹ / b⁻¹)⁻¹ = a / b :=
 by simp
 
-@[simp] lemma sub_sub_cancel (a b : G) : a - (a - b) = b := sub_sub_self a b
+@[simp, to_additive]
+lemma div_div_cancel (a b : G) : a / (a / b) = b := div_div_self' a b
 
-@[simp] lemma sub_sub_cancel_left (a b : G) : a - b - a = -b := by simp
+@[to_additive sub_eq_neg_add]
+lemma div_eq_inv_mul' (a b : G) : a / b = b⁻¹ * a :=
+by rw [div_eq_mul_inv, mul_comm _ _]
 
-lemma sub_eq_neg_add (a b : G) : a - b = -b + a :=
-by rw [sub_eq_add_neg, add_comm _ _]
+@[simp, to_additive]
+lemma div_div_cancel_left (a b : G) : a / b / a = b⁻¹ := by simp
 
-theorem neg_add' (a b : G) : -(a + b) = -a - b :=
-by rw [sub_eq_add_neg, neg_add a b]
+@[to_additive]
+theorem inv_mul' (a b : G) : (a * b)⁻¹ = a⁻¹ / b :=
+by rw [div_eq_mul_inv, mul_inv a b]
 
-@[simp]
-lemma neg_sub_neg (a b : G) : -a - -b = b - a :=
-by simp [sub_eq_neg_add, add_comm]
+@[simp, to_additive]
+lemma inv_div_inv (a b : G) : a⁻¹ / b⁻¹ = b / a :=
+by simp [div_eq_inv_mul', mul_comm]
 
-lemma eq_sub_iff_add_eq' : a = b - c ↔ c + a = b :=
-by rw [eq_sub_iff_add_eq, add_comm]
+@[to_additive eq_sub_iff_add_eq']
+lemma eq_div_iff_mul_eq'' : a = b / c ↔ c * a = b :=
+by rw [eq_div_iff_mul_eq', mul_comm]
 
-lemma sub_eq_iff_eq_add' : a - b = c ↔ a = b + c :=
-by rw [sub_eq_iff_eq_add, add_comm]
+@[to_additive]
+lemma div_eq_iff_eq_mul' : a / b = c ↔ a = b * c :=
+by rw [div_eq_iff_eq_mul, mul_comm]
 
-@[simp]
-lemma add_sub_cancel' (a b : G) : a + b - a = b :=
-by rw [sub_eq_neg_add, neg_add_cancel_left]
+@[simp, to_additive add_sub_cancel']
+lemma mul_div_cancel''' (a b : G) : a * b / a = b :=
+by rw [div_eq_inv_mul', inv_mul_cancel_left]
 
-@[simp]
-lemma add_sub_cancel'_right (a b : G) : a + (b - a) = b :=
-by rw [← add_sub_assoc, add_sub_cancel']
+@[simp, to_additive]
+lemma mul_div_cancel'_right (a b : G) : a * (b / a) = b :=
+by rw [← mul_div_assoc, mul_div_cancel''']
 
-@[simp] lemma sub_add_cancel' (a b : G) : a - (a + b) = -b :=
-by rw [← neg_sub, add_sub_cancel']
+@[simp, to_additive sub_add_cancel']
+lemma div_mul_cancel'' (a b : G) : a / (a * b) = b⁻¹ :=
+by rw [← inv_div', mul_div_cancel''']
 
 -- This lemma is in the `simp` set under the name `add_neg_cancel_comm_assoc`,
```

Comment:
Does the multiplicative version exist there too? If so, can you update this comment?

Reply:
@eric-wieser to be honest I don't understand what that comment means. The only other references to a `add_neg_cancel_comm_assoc` lemma that I can find in the git history are:

https://github.com/leanprover-community/mathlib/commit/f87f8523b997178559df2088a299543f3833df54#diff-2544fc37e1f35d17e2712c0d60ca59c6dc5a2e31a359c8a4acdfb07020d464f3L260

and 

https://github.com/leanprover-community/mathlib/commit/677ae9553749e4f32bf1b482df1543c09e597626#diff-2544fc37e1f35d17e2712c0d60ca59c6dc5a2e31a359c8a4acdfb07020d464f3R260

Reply:
`add_neg_cancel_comm_assoc` indeed exists in `algebra.group.commute`.  A `mul_inv_cancel_comm_assoc` is also there. It looks like the `algebra.group.commute` file has a lot of lemmas which are duplicates of the ones listed here.

I don't know why those results are in two different places, and declared as `simp` only in `commute` but not here in `basic` (the `commute` file transitively imports `basic`). Apparently somebody knew about this, so this might be intentional? I don't see the purpose.

Reply:
```suggestion
-- This lemma is in the `simp` set under the name `mul_inv_cancel_comm_assoc`,
-- along with the additive version `add_neg_cancel_comm_assoc`,
```

Reply:
Ah, of course. I was searching `algebra.group.commute` for `add_neg_cancel_comm_assoc` rather than `mul_inv_cancel_comm_assoc`.